### PR TITLE
CXX-693: allow findAndModifys to take write concern

### PR DESCRIPTION
**Pull request by Machyne**

### CODE REVIEW

**Comment by Machyne** [comment]
_the comment from 6030ce6 should have refferenced #334, not this_

File: src/mongocxx/collection.hpp [hunk 1]
```diff
@@ -350,6 +350,10 @@ class MONGOCXX_API collection {
     /// @return The original or replaced document.
     /// @throws exception::write if the operation fails.
     ///
+    /// @note
```
Comment (acmorrow) [inline src/mongocxx/collection.hpp]: Err... that seems sort of contrary to the spirit of the ticket, which I think was to actually support it?

**Comment by samantharitter** [comment]
LGTM

File: src/mongocxx/collection.hpp (hunk 1 above)
Comment (Machyne) [inline src/mongocxx/collection.hpp]: Right now, this is the only thing possible with the C driver.
We had a bit of discussion [on CDRIVER-945](https://jira.mongodb.org/browse/CDRIVER-945) about this.

Comment (acmorrow) [inline src/mongocxx/collection.hpp]: I see. I don't think that C driver can reasonably decline both this request and the "copy a collection/database" request. Might want to point that out to them.

**Comment by acmorrow** [comment]
I'm fine with committing the comments, but I don't want to resolve the CXX ticket, since we actually still can't do that. Lets add the comments, but leave the ticket open, and move it back to 'open' state, and mark it as waiting on dependency. Leave 0.3.0 as the fix version, and we can address it later, either way.

**Comment by acmorrow** [comment]
LGTM for comment changes.

**Pull request merged**